src/application.c

```diff
@@ -290,6 +300,7 @@ void copy_clicked_handler(GtkWidget *widget, struct swappy_state *state) {
   // Commit a potential paint (e.g. text being written)
   commit_state(state);
   clipboard_copy_drawing_area_to_selection(state);
```

Comment:
Put the same logic inside `clipboard_copy_drawing_area_to_selection`.

i.e:

```suggestion
  if (state->config->early_exit) {
    gtk_main_quit();
  }
```